src/parse/asp/builtins.go

```diff
@@ -80,6 +77,10 @@ func registerBuiltins(s *scope) {
 		"get":        setNativeCode(s, "config_get", configGet),
 		"setdefault": s.Lookup("setdefault").(*pyFunc),
 	}
+	setNativeCode(s, "git_branch", execGitBranch)
```

Comment:
I wonder if, rather than having four builtins here, if there was just one for running a git command and then four wrappers around it in the builtin rules that invoke it with different arguments. Thoughts?

Reply:
I thought about that, too.  Something like a `git()` command with a list of sub-commands, like:

```python
def git(command:list, args:list=None) -> str:
    pass
```

Where `command` is a whitelist of commands.  Maybe?  That seemed like a huge can of worms to me.  In most cases, you want to interrogate a git repository and report out various bits of information.  Judging by prior art, in the common case it looks like most people seem to be interested in a handful of commands and pieces of information, not all of the functionality available to them via `git()`.  From a usability perspective, it seemed like there were maybe only 3-4 things that people would want to regularly use (see [`govvv`](https://github.com/ahmetb/govvv) and [`goreleaser`](https://github.com/goreleaser/goreleaser/blob/e08cc5003097b71b17e0ec86c1354de9792832c7/www/content/build.md#L46)).

Reply:
Yeah, I sort of imagined exposing only the specific commands (`git_commit` and so forth), although I don't *think* there is a suitable mechanism for hiding the underlying `git` function.

I guess that's an implementation detail. Can always change that later since we're agreed on the exposed interface.